src/main/java/com/example/hop_oasis/decoder/ExtractImageName.java

```diff
@@ -0,0 +1,14 @@
+package com.example.hop_oasis.decoder;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ExtractImageName {
+    public String extractName(String name) {
+        if (name.startsWith("http://") || name.startsWith("https://")) {
+            return name.substring(name.lastIndexOf('/') + 1);
+        }else {
```

Comment:
why my aproach is bad?

Reply:
because you can do it way simpler, basically in one line, without any `ifs`
if `name` doesn't contain any '/' you will have an array with only one element
`return name.split("/")[array.length - 1];`

Reply:
Well, to be honest, my approach is also bad. 
A client will send you the full URL for the image:
`https://bucket-name/path/to/image/123.jpg`
In this case, you must preserve the whole path to the image.
In your approach, you'll get only `123.jpg` part, but you need `/path/to/image/123.jpg`.
I suggest you to take everything after schema(https://) and use it as a parameter to delete an image.

```
String[] array = name.split("://");
return array[array.length - 1];
```